cli/src/cast.rs

```diff
@@ -317,6 +327,67 @@ async fn main() -> eyre::Result<()> {
                     }
                 }
             }
+            WalletSubcommands::Vanity { starts_with, ends_with, max_threads } => {
+                let mut regexs = vec![];
+                if let Some(prefix) = starts_with {
+                    hex::decode(&prefix).expect("invalid prefix hex provided");
+                    regexs.push(format!(r"^{}", prefix));
+                }
+                if let Some(suffix) = ends_with {
+                    hex::decode(&suffix).expect("invalid suffix hex provided");
+                    regexs.push(format!(r"{}$", suffix));
+                }
+
+                assert!(
+                    regexs.iter().map(|p| p.len() - 1).sum::<usize>() <= 40,
+                    "vanity patterns length exceeded. cannot be more than 40 characters",
+                );
+
+                let regex = RegexSet::new(regexs)?;
+
+                let max_threads = match max_threads {
+                    Some(num) => {
+                        assert!(num > 0, "number of threads cannot be 0");
+                        num
+                    }
```

Comment:
don't feel comfortable without having an upper bound for the number of maximum threads, open to suggestions

Reply:
there's the num_cpu crate which is usually used for this kind of stuff,
alternative suggestion is to rewrite this rewrite this as a rayon parallel iterator

basically an endless iterator over random wallets, turn this into a parallel iterator with rayon's .par_iter or .par_bridge and then probably call find_first 
https://docs.rs/rayon/latest/rayon/iter/trait.ParallelIterator.html#method.find_first

this would remove all of the complexity of managing the threads yourself.

Reply:
How would you do an endless iterator? `std::iter::cycle(1)`?

Reply:
https://doc.rust-lang.org/std/iter/fn.repeat_with.html

But I wonder if the borrow checker will complain for `LocalWallet::new(&mut thread_rng())`

Reply:
@mattsse gonna try this out, thanks!

Reply:
couldn't make it work with `par_iter`, but `par_bridge` worked like a charm. imho looks much cleaner, thanks @mattsse!